functions.js

```diff
@@ -0,0 +1,106 @@
+            let routes = pairs.map(([originIndex, destinationIndex]) => {
+                const originStation = origins[originIndex];
+                const destinationStation = destinations[destinationIndex];
+
+                return {
+                    line: originStation.line,
+                    start: Math.min(originStation.position, destinationStation.position),
+                    end: Math.max(originStation.position, destinationStation.position)
+                }
+            })
+
+            //Query Journey Data
+            const journeys = []
+            for (const route of routes) {
+                let journey = await getSingleLineJourney(route.line, route.start, route.end);
+                journeys.push(journey);
+            }
+
+            //Summarise data to provide information in the format of end Route Card
+            const journeySummaries = []
+
+            for (const journey of journeys) {
+
+                let journeyArrayLength = journey.length - 1
+
+                if (journey[0].code === originSelection) {
+                    const from = journey[0].name
+                    const to = journey[journeyArrayLength].name
+                    const line = journey[0].line
+                    const stationBreakdown = journey.map(station => station.name)
+                    journeySummaries.push({
+                        from: from, to: to, line: line, stations: stationBreakdown
+                    })
+                } else {
+                    const from = journey[journeyArrayLength].name
+                    const to = journey[0].name
+                    const line = journey[journeyArrayLength].line
+                    const stationBreakdown = journey.map(station => station.name).reverse()
+                    journeySummaries.push({
+                        from: from, to: to, line: line, stations: stationBreakdown
+                    })
+                }
+            }
+
+            // Provide response back to the request
+            if (journeys.length > 0) {
+                res.status(200).json({
+                    message: "Successfully Retrieved Journeys.", data: journeys, summary: journeySummaries
+                })
+            } else {
+                res.status(204).json({message: "No Valid Journeys", data: []})
+            }
+        }
+    } catch (e) {
+        res.status(500).json({message: "Unexpected Error", data: []})
+    }
+}
+
+const getStationInstances = async (stationCode) => {
+    const db = await dbConnection
+    return db.query('SELECT `id`, `code`, `name`, `timeToPrev`, `timeToNext`, `zone`, `line`, `position` FROM `stations` WHERE `code` = ?', [stationCode])
+}
+
+const getSingleLineJourney = async (line, start, end) => {
+    const db = await dbConnection
+    return db.query('SELECT `id`, `code`, `name`, `timeToPrev`, `timeToNext`, `zone`, `line`, `position` FROM `stations` WHERE `line` = ? AND `position` >= ? AND `position` <= ?', [line, start, end])
+}
```

Comment:
Though this will still work, it's usually better to move function definitions to before their usages